tools/osbuild-depsolve-dnf5

```diff
@@ -157,6 +157,22 @@ class Solver():
                 self._dnfrepo(repo, exclude_pkgs)
 
             if root_dir:
+                # This sets the varsdir to ("{root_dir}/etc/yum/vars/", "{root_dir}/etc/dnf/vars/") for custom variable
+                # substitution (e.g. CentOS Stream 9's $stream variable)
+                conf.installroot = root_dir
+                conf.varsdir = (os.path.join(root_dir, "etc/dnf/vars"), os.path.join(root_dir, "usr/share/dnf5/vars.d"))
```

Comment:
In dnf4, we have:
```python
base.conf.substitutions.update_from_etc(root_dir)
```

which will load any vars from `<root_dir>/etc/dnf/vars` and `<root_dir>/etc/yum/vars`.  This is verified in our test.
From the dnf5 docs, I got the impression that setting the installroot would make it load vars from `<installroot>/etc/dnf/vars` and `<installroot>/usr/share/dnf5/vars.d`, but this isn't working.